compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -302,6 +305,20 @@ private void writeFunction(ByteBuf buf, BIRTypeWriter typeWriter, BIRInstruction
         buf.writeBytes(birbuf.nioBuffer().array(), 0, length);
     }
 
+    private void writeFunctionsGlobalVarDependency(ByteBuf buf, BIRNode.BIRFunction birFunction) {
+        List<Integer> globalVarBuf = new LinkedList<>();
+
+        for (BIRNode.BIRVariableDcl var : birFunction.dependentGlobalVars) {
+            if (var == null) {
```

Comment:
can this be null at any point?

Reply:
Earlier there was another bug collecting constants as well. I also fixed that issue along with this but forgot to remove it here. Fixed it.